pkg/handler/status.go

```diff
@@ -1,14 +1,74 @@
 package handler
 
 import (
+	"context"
 	"net/http"
-
-	"github.com/sirupsen/logrus"
+	"strings"
 )
 
-func Status(w http.ResponseWriter, _ *http.Request) {
-	_, err := w.Write([]byte("OK"))
-	if err != nil {
-		logrus.Errorf("could not write response: %v", err)
+type Status struct {
+	IsAllowProm         bool `yaml:"isAllowProm" json:"isAllowProm"`
+	PromNamespacesCount int  `yaml:"promNamespacesCount" json:"promNamespacesCount"`
+	IsAllowLoki         bool `yaml:"isAllowLoki" json:"isAllowLoki"`
+	LokiNamespacesCount int  `yaml:"lokiNamespacesCount" json:"lokiNamespacesCount"`
+	IsLokiReady         bool `yaml:"isLokiReady" json:"isLokiReady"`
+	IsConsistent        bool `yaml:"isConsistent" json:"isConsistent"`
+}
+
+func (h *Handlers) Status(ctx context.Context) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		params := r.URL.Query()
+		namespace := params.Get(namespaceKey)
+		isDev := namespace != ""
+
+		status := Status{
+			IsAllowProm: h.Cfg.IsPromEnabled(),
+			IsAllowLoki: h.Cfg.IsLokiEnabled(),
+		}
+
+		if status.IsAllowProm {
+			promClients, err := newPromClients(h.Cfg, r.Header, namespace)
+			if err != nil {
+				writeError(w, http.StatusInternalServerError, err.Error())
+				return
+			}
+			// get namespaces using Prom
+			promNamespaces, code, err := h.getNamespacesValues(ctx, promClients, isDev)
+			if err != nil {
+				writeError(w, code, "Error while fetching label namespace values from Prometheus: "+err.Error())
```

Comment:
Here and below: the downside of throwing an error is that on the frontend side we loose the information about what is enabled (loki and/or prom). Since this is a status check, what about, instead of throwing an error, just set `false` to IsLoki/PromReady + add the error as a message embedded in the Status result?

Reply:
To illustrate:
![image](https://github.com/user-attachments/assets/963a3911-79c4-48ed-a8d2-2d60e13baf79)

Here on the frontend side we could determine what to display based on if it's a loki or a prometheus permission error, but we don't know that due to the status endpoint failing to provide the status.